.github/workflows/static-analysis.yaml

```diff
@@ -0,0 +1,28 @@
+---
+name: Static Analysis
+on: [push, pull_request]
+permissions: read-all
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+      - id: goversion
+        run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+      - uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
+        with:
+          go-version: ${{ steps.goversion.outputs.goversion }}
+      - run: |
+          set -euo pipefail
+          make verify
+      - run: |
+          set -euo pipefail
```

Comment:
Nit: Line breaks in this file don't match one for `release-3.4` or `main`. Could we please make it the same so the files are easy to diff in future if needed.

Reply:
Good catch @jmhbnz thanks, I probably missed something while copying over the content. 
Done!